server/src/main/java/com/thoughtworks/go/config/update/UpdateTemplateAuthConfigCommand.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config.update;
+
+import com.thoughtworks.go.config.CruiseConfig;
+import com.thoughtworks.go.config.PipelineTemplateConfig;
+import com.thoughtworks.go.config.TemplatesConfig;
+import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.EntityHashingService;
+import com.thoughtworks.go.server.service.SecurityService;
+import com.thoughtworks.go.server.service.result.LocalizedOperationResult;
+
+public class UpdateTemplateAuthConfigCommand extends UpdateTemplateConfigCommand {
+
+    public UpdateTemplateAuthConfigCommand(PipelineTemplateConfig templateConfig, Username currentUser, SecurityService securityService, LocalizedOperationResult result, String md5, EntityHashingService entityHashingService) {
+        super(templateConfig, currentUser, securityService, result, md5, entityHashingService);
+    }
+
+    @Override
+    public void update(CruiseConfig modifiedConfig) throws Exception {
+        PipelineTemplateConfig existingTemplateConfig = findAddedTemplate(modifiedConfig);
+        TemplatesConfig templatesConfig = modifiedConfig.getTemplates();
+        existingTemplateConfig.setAuthorization(templateConfig.getAuthorization());
+        modifiedConfig.setTemplates(templatesConfig);
```

Comment:
I couldn't understand the need to get templates config and set it back again, is this required?

Reply:
No, templates need not be set. I'll change it.